GraphsV2/TreeGraph/index.js

```diff
@@ -0,0 +1,598 @@
+        nodeUpdate.transition()
+            .duration(d => d.data.loaded ? 0 : duration)
+            .attr('transform', d => `translate(${d.y}, ${d.x})`);
+
+        // Remove any exiting nodes
+        node.exit().transition()
+            .duration(duration)
+            .attr('transform', () => `translate(${source.y}, ${source.x})`)
+            .remove();
+
+        nodeUpdate.select('rect')
+            .style('fill', d => d.data.clicked ? rectNode.selectedBackground : rectNode.defaultBackground);
+    }
+
+    const renderRectNode = d => {
+        const {
+            renderNode,
+            store,
+            ...rest
+        } = props;
+
+        if (!isFunction(renderNode)) return '<div/>';
+
+        const rectColorText = d.data.clicked ? rectNode.selectedTextColor : rectNode.defaultTextColor;
+
+        return renderNode(
+            {
+                data: d.data,
+                textColor: rectColorText,
+                nodeId: `node-${d.id}`,
+                store,
+                ...rectNode,
+                onItemRender: () => (rest)
+            }
+        );
+    }
+
+    const updateLinks = (source, links) => {
+        // update graph
+        const svg = getGraphContainer();
+
+        // ****************** links section ***************************
+        // Update the links...
+        const link = svg.selectAll('path.link').data(links, d => d.id);
+
+        // // Enter any new links at the parent's previous position.
+        const linkEnter = link.enter().insert('path', 'g')
+            .attr('class', 'link')
+            .attr('d', d => diagonal(d))
+            .attr('stroke-width', linksSettings.stroke.width)
+
+        normalArrow(svg, linksSettings);
+        coloredArrow(svg, linksSettings);
+
+        // UPDATE
+        const linkUpdate = linkEnter.merge(link);
+
+        linkUpdate.style('stroke', d => {
+            return d.data.clicked ? linksSettings.stroke.selectedColor : linksSettings.stroke.defaultColor;
+        })
+
+        linkUpdate.transition()
+            .duration(d => d.data.loaded ? 0 : duration)
+            .attr('d', d => diagonal(d))
+            .attr('marker-start', d => d.data.clicked ? 'url(#colored-arrow)' : 'url(#normal-arrow)');
+
+        // Remove any exiting links
+        link.exit().transition()
+            .duration(duration)
+            .attr('d', d => diagonal(d))
+            .remove();
+    }
+
+    const normalArrow = (svg, linksSettings) => {
+        svg.append('svg:defs').append('marker')
+            .attr('id', 'normal-arrow')
+            .attr('viewBox', '0 -5 10 10')
+            .attr('refX', 0)
+            .attr('refY', 0)
+            .attr('markerWidth', 6)
+            .attr('markerHeight', 6)
+            .attr('orient', 'auto')
+            .attr('fill', linksSettings.stroke.defaultColor)
+            .append('path')
+            .attr('d', 'M10,-5L0,0L10,5');
+    }
+
+    const coloredArrow = (svg, linksSettings) => {
```

Comment:
this does not depend on any of the component state, let's move it out of component.